audition/thresholding.py

```diff
@@ -0,0 +1,145 @@
+        for metric_filter in self._metric_filters:
+            passing &= set(filter_func(
+                _of_metric(df, metric_filter),
+                metric_filter
+            )['model_group_id'])
+        return passing
+
+    def model_groups_above_min(self, df):
+        """Return the model groups in the dataframe that are above the
+            currently-configured minimum value
+
+        Args:
+            df (pandas.DataFrame): A set of rows in the format given by
+                audition.DistanceFromBestTable.as_dataframe
+        Returns: (set) The model group ids above the minimum value for each metric
+        """
+        return self._filter_model_groups(df, _above_min)
+
+    def model_groups_close_to_best(self, df):
+        """Return the model groups in the dataframe that are close enough to
+            the best value according to current metric filter configuration
+
+        Args:
+            df (pandas.DataFrame): A set of rows in the format given by
+                audition.DistanceFromBestTable.as_dataframe
+        Returns: (set) The model group ids close to the best value for each metric
+        """
+        return self._filter_model_groups(df, _close_to_best)
+
+    def model_groups_passing_rules(self):
+        """Return the model groups passing both the close-to-best and
+            above-min checks based on the current filters.
+
+        Works by ensuring that a model group passes the close-to-best check
+        for at least one train end time, and that is passes the above-min
+        check for *all* train end times. Model groups must pass both
+        of these checks.
+
+        Returns: (set) The passing model group ids
+        """
+        below_min_model_groups = set(self._initial_model_group_ids)
+        close_to_best_model_groups = set()
+        for train_end_time in self.train_end_times:
+            df_as_of = self.distance_from_best_table.dataframe_as_of(
+                model_group_ids=self._initial_model_group_ids,
+                train_end_time=train_end_time,
+            )
+            close_to_best = self.model_groups_close_to_best(df_as_of)
+            logging.info(
+                'Found %s model groups close to best for %s',
+                len(close_to_best),
+                train_end_time
+            )
+            close_to_best_model_groups |= close_to_best
+
+            below_min = self.model_groups_above_min(df_as_of)
+            logging.info(
+                'Found %s model groups below min for %s',
+                len(below_min),
+                train_end_time
+            )
+            below_min_model_groups &= below_min
+
+        total_model_groups = close_to_best_model_groups & below_min_model_groups
+        logging.info(
+            'Found %s total model groups past threshold',
+            len(total_model_groups)
+        )
+        return total_model_groups
+
+    def update_filters(self, new_metric_filters):
```

Comment:
having to call `update_filters()` immediately after initializing the object feels a little clunky; it probably makes sense to requiring passing an initial filter set to the constructor